From 8400d4700796bfb377aa386b6a04f6c7d1680906 Mon Sep 17 00:00:00 2001
From: Oliver Steffen <osteffen@redhat.com>
Date: Mon, 26 Oct 2026 13:53:31 +0100
Subject: [PATCH 1/3] Disable TDES algorithm for bare-metal SVSM

The bare-metal OpenSSL build does not include DES support. Disable
ALG_TDES and guard TDES-specific code behind #if ALG_TDES to allow
building without DES headers and symbols.
---
 .../TpmConfiguration/TpmConfiguration/TpmProfile_Common.h       | 2 +-
 src/tpm2/TPMCmd/tpm/cryptolibs/Ossl/include/Ossl/TpmToOsslSym.h | 2 ++
 src/tpm2/TPMCmd/tpm/include/private/CryptSym.h                  | 2 ++
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/tpm2/TPMCmd/TpmConfiguration/TpmConfiguration/TpmProfile_Common.h b/src/tpm2/TPMCmd/TpmConfiguration/TpmConfiguration/TpmProfile_Common.h
index 9bf1c12..c7b847e 100644
--- a/src/tpm2/TPMCmd/TpmConfiguration/TpmConfiguration/TpmProfile_Common.h
+++ b/src/tpm2/TPMCmd/TpmConfiguration/TpmConfiguration/TpmProfile_Common.h
@@ -79,7 +79,7 @@
 #define     CAMELLIA_192                (YES * ALG_CAMELLIA) /* since libtpms v0.10 stateFormatLevel 4 */
 #define     CAMELLIA_256                (YES * ALG_CAMELLIA)
 
-#define ALG_TDES                        ALG_YES 		/* libtpms added begin; enabled */
+#define ALG_TDES                        ALG_NO 		/* disabled for bare-metal SVSM */
 
 #define     TDES_128                    (ALG_TDES && YES)
 #define     TDES_192                    (ALG_TDES && YES)	/* libtpms added end */
diff --git a/src/tpm2/TPMCmd/tpm/cryptolibs/Ossl/include/Ossl/TpmToOsslSym.h b/src/tpm2/TPMCmd/tpm/cryptolibs/Ossl/include/Ossl/TpmToOsslSym.h
index 90d434a..21f3502 100644
--- a/src/tpm2/TPMCmd/tpm/cryptolibs/Ossl/include/Ossl/TpmToOsslSym.h
+++ b/src/tpm2/TPMCmd/tpm/cryptolibs/Ossl/include/Ossl/TpmToOsslSym.h
@@ -86,7 +86,9 @@ typedef void (*TpmCryptSetSymKeyCall_t)(const BYTE* in, BYTE* out, void* keySche
 #define TpmCryptDecryptAES AES_decrypt
 #define tpmKeyScheduleAES  AES_KEY
 
+#if ALG_TDES
 #include "TpmToOsslSymTDES.h" // libtpms added
+#endif
 
 //***************************************************************
 //** Links to the OpenSSL SM4 code
diff --git a/src/tpm2/TPMCmd/tpm/include/private/CryptSym.h b/src/tpm2/TPMCmd/tpm/include/private/CryptSym.h
index 4ed786e..28ebefc 100644
--- a/src/tpm2/TPMCmd/tpm/include/private/CryptSym.h
+++ b/src/tpm2/TPMCmd/tpm/include/private/CryptSym.h
@@ -48,7 +48,9 @@
 typedef union tpmCryptKeySchedule_t {
     FOR_EACH_SYM_WITHOUT_TDES(KEY_SCHEDULE)	/* libtpms changed from FOR_EACH_SYM */
 
+#if ALG_TDES
     tpmKeyScheduleTDES  tdes[3];		/* libtpms added */
+#endif
 
 #if SYMMETRIC_ALIGNMENT == 8
     uint64_t alignment;
-- 
2.56.0

